### ISSUE TITLE
RuntimeError: imageio.ffmpeg.download() has been deprecated. Use 'pip install imageio-ffmpeg' instead.'

### ISSUE THREAD

**Issue by devalladares** [mikaelalafriz/lucid-sonic-dreams#41]
[](url)
<img width="962" alt="Screenshot 2022-10-14 at 11 29 46 PM" src="https://user-images.githubusercontent.com/60202683/195967003-321e3460-e513-475f-ab7c-1fed8f98ff3b.png">

Code's not running anymore :( Was one of my favourite notebooks, please let me know if and how it can be fixed!

**Comment by deeplearn-art** [comment]
Try adding 
`!pip install imageio==2.4.1`
before installing lucidsonicdreams

However there will be more errors...

**Comment by devalladares** [comment]
@deeplearn-art thank you so much!! While it processed that stage, when I went to the next step to process the video, I got another error that also seems sound related —
```

Setting up TensorFlow plugin "fused_bias_act.cu": Failed!
WARNING:py.warnings:/usr/local/lib/python3.7/dist-packages/librosa/core/audio.py:165: UserWarning: PySoundFile failed. Trying audioread instead.
  warnings.warn("PySoundFile failed. Trying audioread instead.")

Preparing audio...
WARNING:py.warnings:/usr/local/lib/python3.7/dist-packages/librosa/core/audio.py:165: UserWarning: PySoundFile failed. Trying audioread instead.
  warnings.warn("PySoundFile failed. Trying audioread instead.")
```

Do let me know if you have an idea how to go about this thank you so much! 

<img width="1308" alt="Screen Shot 2022-10-17 at 4 54 21 PM" src="https://user-images.githubusercontent.com/60202683/196281281-14080f3d-738a-43e8-93e0-9357f2cd3ee7.png">

**Comment by deeplearn-art** [comment]
Yes, that issue has been reported before. It used to happen in local installation but not in colab.
Unfortunately there is no obvious fix, I think the packages have changed and no one has saved a working configuration. 

The Deep Music Visualizer still works though
https://colab.research.google.com/github/Ease78/Deep-Music-Visualizer-Google-Colab/blob/master/DeepMusicVisualizer.ipynb

**Comment by rwconsultancy** [comment]
I managed to get past the above problem, including the audioread one that the OP mentions, by using this sequence:

! pip install moviepy
! pip3 install imageio==2.4.1
! pip install --upgrade imageio-ffmpeg
! pip install lucidsonicdreams
from google.colab import files
from lucidsonicdreams import LucidSonicDream

I can't claim any credit for it though; I just found it on in a dark corner of Reddit.

That held for a little while but then, about 2 weeks ago, something else changed in the Colab environment and I started getting a brand new error:

![opera_JwttwdkQjW](https://user-images.githubusercontent.com/79518075/196528416-9c63eb72-3180-4b66-bc25-048a0205f0d8.png)

Interestingly, if I follow the instructions to use the BigGan function, instead of using a StyleGan style, it still works - but I'd really like use StyleGan again if at all possible.

**Comment by udiallo** [comment]
I have also tried to get the package running again in the last few days but also failed because of the assertion error.

Can you explain how you used the BigGan variant or where to find info about it?

**Comment by rwconsultancy** [comment]
It's here, down at the bottom:

https://colab.research.google.com/drive/1Y5i50xSFIuN3V4Md8TB30_GOAtts7RQD

What I found is that running the BigGan code there results in an error that demands you restart the runtime. You then have to run everything from the top (ie from '! pip install moviepy 'in the snippet I posted above, and then the BigGan stuff), all over again and then it's happy.

**Comment by udiallo** [comment]
Okay, so I had a `No GPU available` at first, but after changing the runtime type to GPU it worked. Thank you.

Anyways, I would really like to get the StyleGan back to work. It should be possible to include the StyleGan model in the same way as the BigGan model, or even using some different StyleGan version, such as [this pytorch StyleGan](https://github.com/NVlabs/stylegan2-ada-pytorch). Any opinions on these ideas?

**Comment by rwconsultancy** [comment]
I really don't enough about this stuff to comment unfortunately. One big question though is: what changed in Colab to suddenly break this? I followed some instructions to downgrade the Python version to see if that would undo the change but, when all it did was open other big can of errors, I abandoned that line of enquiry. Any clever people out there with any ideas?

**Comment by deeplearn-art** [comment]
> I really don't enough about this stuff to comment unfortunately. One big question though is: what changed in >Colab to suddenly break this?

The required packages are listed in setup.py, I think it likely that one or more have been updated by Colab and because of that, the notebook is broken. 

To avoid this, it is customary to specify working version of the modules, but apart from tensorflow==1.15 this has not been done. 

If someone has this working locally, maybe they might know the correct version numbers.

**Comment by ristmatt** [comment]
Is there any way this notebook could run on kaggle?

**Comment by cnatic** [comment]
would also love to see this run again.

**Comment by udiallo** [comment]
Unfortunately, I also have no GPU locally with which I could try to find the broken packages. I heard it is also possible to use Google VM's for free (to a certain extend). Does anybody have experiences with Lucid Sonic Dreams on a Google VM? Maybe it is worth trying it to at least find the broken packages and fix them on Colab then.

**Comment by rwconsultancy** [comment]
Im lost for ideas Im afraid  I really dont know much about any of this
stuff! One question of course is: what changed in the Colab environment a
couple of weeks ago that stopped it working? I tried following some
instruction to downgrade my Colab to Python 3.6 but that didnt bring it
back (in fact it just created a whole heap of new errors).

 

----- Original Message -----

Subject: Re: [mikaelalafriz/lucid-sonic-dreams] RuntimeError:
imageio.ffmpeg.download() has been deprecated. Use 'pip install
imageio-ffmpeg' instead.' (Issue #41)

Date: Wed, 19 Oct 2022 14:31:58 -0700

From: Diaoulé Diallo ***@***.***
***@***.***> >

 

Okay, so I had a `No GPU available` at first, but after changing the runtime
type to GPU it worked. Thank you.

 

Anyways, I would really like to get the StyleGan back to work. It should be
possible to include the StyleGan model in the same way as the BigGan model,
or even using some different StyleGan version, such as [this pytorch
StyleGan](https://github.com/NVlabs/stylegan2-ada-pytorch). Any opinions on
these ideas?

 

 

 

-- 

Reply to this email directly or view it on GitHub:

https://github.com/mikaelalafriz/lucid-sonic-dreams/issues/41#issuecomment-1
284595810

You are receiving this because you commented.

 

Message ID:
***@***.***
***@***.***> >
